src/jet_client.rs

```diff
@@ -144,224 +151,331 @@ impl Future for JetMsgReader {
                                 }
                             } else {
-                                error!("No candidate found for an association version 1. Should never happen.");
-                                status_code = StatusCode::INTERNAL_SERVER_ERROR;
+                                unreachable!("No candidate found for an association version 1. Should never happen.");
                             }
                         }
                         (2, 2) => {
                             if let Some(candidate) = association.get_candidate_mut(self.request_msg.candidate) {
-                                if candidate.transport_type() == TransportType::Tcp && candidate.state() == CandidateState::Accepted {
-                                    if let Some(transport) = candidate.server_transport() {
-                                        candidate.set_state(CandidateState::Connected);
-                                        candidate.set_client_transport(self.transport.clone());
-
-                                        // The accept request has been received before and a transport is available to open the proxy
-                                        self.server_transport = Some(transport);
-                                        self.candidate = Some(candidate.clone());
-                                        status_code = StatusCode::OK;
-                                    }
+                                if candidate.transport_type() == TransportType::Tcp
+                                    && candidate.state() == CandidateState::Accepted
+                                {
+                                    Some(candidate)
+                                } else {
+                                    None
                                 }
                             } else {
                                 status_code = StatusCode::NOT_FOUND;
+
+                                None
                             }
                         }
                         (association_version, request_version) => {
-                            error!("Invalid version: Association version={}, Request version={}", association_version, request_version);
+                            error!(
+                                "Invalid version: Association version={}, Request version={}",
+                                association_version, request_version
+                            );
+
+                            None
                         }
-                    }
+                    };
 
+                    if let Some(candidate) = candidate {
+                        // The accept request has been received before and a transport is available to open the proxy
+                        if let Some(server_transport) = candidate.take_transport() {
+                            candidate.set_state(CandidateState::Connected);
+
+                            self.server_transport = Some(server_transport);
```

Comment:
We get the transport from the candidate (he is our target server), temporarily save, and also save the association_id, candidate_id, which will be needed for the result.